RandBLAS/trig_skops.hh

```diff
@@ -0,0 +1,538 @@
+#include "RandBLAS/base.hh"
+#include "RandBLAS/exceptions.hh"
+#include "RandBLAS/random_gen.hh"
+#include "RandBLAS/sparse_skops.hh"
+#include "util.hh"
+
+#include <Random123/philox.h>
+#include <blas.hh>
+
+#include <iostream>
+#include <stdio.h>
+#include <stdexcept>
+#include <string>
+#include <sys/_types/_int64_t.h>
+#include <tuple>
+
+#include <math.h>
+#include <typeinfo>
+#include <vector>
+
+#define MAX(a, b) (((a) < (b)) ? (b) : (a))
+#define MIN(a, b) (((a) < (b)) ? (a) : (b))
+
+namespace RandBLAS {
+    // =============================================================================
+    /// WARNING: None of the following functions or overloads thereof are part of the
+    /// public API
+    ///
+
+    // Generates a vector of Rademacher entries using the Random123 library
+    template<SignedInteger sint_t = int64_t, typename state_t = RNGState<DefaultRNG>>
+    state_t generate_rademacher_vector_r123(sint_t* buff, int64_t n, state_t &seed_state) {
+        DefaultRNG rng;
+        auto [ctr, key] = seed_state;
+
+        for (int64_t i = 0; i < n; ++i) {
+            typename DefaultRNG::ctr_type r = rng(ctr, key);
+
+            float rand_value = r123::u01fixedpt<float>(r.v[0]);
+
+            buff[i] = rand_value < 0.5 ? -1 : 1;
+
+            ctr.incr();
+        }
+
+        // Return the updated RNGState (with the incremented counter)
+        return state_t {ctr, key};
+    }
+
+    // Catch-all method for applying the diagonal Rademacher
+    // entries in-place to an input matrix, `A`
+    template<typename T, SignedInteger sint_t = int64_t>
+    void apply_diagonal_rademacher(
+                                bool left, // Pre-multiplying?
+                                blas::Layout layout,
+                                int64_t rows,
+                                int64_t cols,
+                                T* A,
+                                sint_t* diag
+                                ) {
```

Comment:
fix indentation to be consistent with our conventions. e.g., those here: https://github.com/BallisticLA/RandBLAS/blob/main/RandBLAS/skve.hh